### PR TITLE
Syndicate beacon topic now acts only on user

### LINKED ISSUE #20716: Any human mob can be forced to become a traitor
The `betraitor` topic call on the [Syndicate Beacon](https://github.com/tgstation/tgstation/blob/master/code/game/machinery/syndicatebeacon.dm) can be abused to force any human mob in the game to become a traitor.

```dm
/obj/machinery/syndicate_beacon/Topic(href, href_list)
        if(..())
                return
        if(href_list["betraitor"])
                if(charges < 1)
                        src.updateUsrDialog()
                        return
                var/mob/M = locate(href_list["traitormob"])
```

### PR DESCRIPTION
**Pull request by optimumtact**
Fixes #20716
Closes #20719

### CODE REVIEW

File: code/game/machinery/syndicatebeacon.dm [hunk 1]
```diff
@@ -46,6 +46,9 @@
 			src.updateUsrDialog()
 			return
 		var/mob/M = locate(href_list["traitormob"])
+		if(M != usr)
+			usr << "Cut that shit out"
```
Comment (Iamgoofball) [inline code/game/machinery/syndicatebeacon.dm]: alert the 'mins

**Comment by PKPenguin321** [comment]
The superior PR

**Comment by PKPenguin321** [comment]
:(

**Comment by optimumtact** [comment]
This is so rarely used that there is not much point shedding a tear over it.

**Pull request closed without merging**